homeassistant/components/binary_sensor/template.py

```diff
@@ -143,6 +172,33 @@ def _async_render(self):
                 return
             _LOGGER.error("Could not render template %s: %s", self._name, ex)
 
+        for property_name, template in (
+                ('_icon', self._icon_template),
+                ('_entity_picture', self._entity_picture_template)):
+            if template is None:
+                continue
+
+            try:
+                setattr(self, property_name, template.async_render())
+            except TemplateError as ex:
+                friendly_property_name = property_name[1:].replace('_', ' ')
+                if ex.args and ex.args[0].startswith(
+                        "UndefinedError: 'None' has no attribute"):
+                    # Common during HA startup - so just a warning
+                    _LOGGER.warning('Could not render %s template %s,'
+                                    ' the state is unknown.',
+                                    friendly_property_name, self._name)
+                    return
+
+                try:
```

Comment:
Not sure about this last piece of code. What does it do? 🤔 Because the super class doesn't have a property name called `_icon`. I think that in this case you should probably set it to `None` or leave it as the previous value?

Reply:
What are the chances this was done in case the parent ever changed and got icon or entity_picture variables?  SensorTemplate parent is Entity and BinarySensorTemplate parent is BinarySensorDevice (then Entity)

Reply:
I looked a bit at this and it seems likely that this is a thinko in the original implementation and an example that error handling is rarely well-tested. So I think the `super` stuff can just be removed 👍

I do wonder, though: rather than copy/paste, could this be implemented in a more central place?

Reply:
A little more digging and I found this code is duplicate in cover.template, light.template, sensor.template, switch.template and this patch would make it binary_sensor.template as well.  

Is there a issue merging this PR, then opening another issue to clean up the duplicate template stuff?

Reply:
Doing the cleanup in a separate PR is fine but I think we should not merge the `super` stuff now that we know it is wrong.